Source/Core/VideoCommon/FreeLookUDPServer.cpp

```diff
@@ -0,0 +1,113 @@
+// Copyright 2021 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "VideoCommon/FreeLookUDPServer.h"
+
+#include "Common/MathUtil.h"
+#include "VideoCommon/FreeLookUDPSpec.h"
+
+namespace
+{
+template <typename T>
+T GetMessage(const u8*& data)
+{
+  T msg;
+  memcpy(&msg, data, sizeof(T));
+  data += sizeof(T);
+  return msg;
+}
+}  // namespace
+
+FreeLookUDPServer::FreeLookUDPServer(std::string address, u16 port)
+{
+  m_socket.setBlocking(false);
+  m_socket.bind(port, address);
+}
+
+void FreeLookUDPServer::Update()
+{
+  std::array<u8, sf::UdpSocket::MaxDatagramSize> buffer;
+
+  sf::IpAddress sender;
+  unsigned short port;
+  std::size_t bytes_received;
+  while (m_socket.receive(buffer.data(), buffer.size(), bytes_received, sender, port) ==
+         sf::Socket::Done)
+  {
+    ProcessReceivedData(buffer.data(), bytes_received);
+  }
+}
+
+void FreeLookUDPServer::ProcessReceivedData(const u8* data, std::size_t packet_size)
+{
+  const std::size_t header_size = sizeof(PacketHeader);
+  if (packet_size < header_size)
+    return;
+
+  PacketHeader header;
+  memcpy(&header, data, header_size);
+
+  if (header.version != FREE_LOOK_PROTO_VERSION)
```

Comment:
If I recall, standard UDP packages already come with a version number, which I guess is 0 or 1, so why don't you officially propose this new version of the standard and implement all of this as a switch based on package version in the InputCommon UDP server?

Reply:
What if someone customized the UDP protocol in the future and used the same version number as you did here? Which then could also have the same packet size.

Reply:
On a general level, what you're describing is something that is discussed when managing any sort of wire transfer.  How much validation can we do?  It's usually a trade off of speed vs ensuring correctness.  At some point though, you have to take whatever you're given and assume it is correct.

If the protocol version is an expected value and the packet size / message type are appropriate, we assume the data is appropriate.  You're right though, we could be given a character array of the same size or 2 floats instead of every double, etc.

I don't know how important it is to do more validation than message type/size and version check.

Reply:
It's fine as it is. My second comment didn't make much sense.